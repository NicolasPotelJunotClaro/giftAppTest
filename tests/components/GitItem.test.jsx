import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GitItem", () => {
  const title = "One punch";
  const url =
    "https://static.wikia.nocookie.net/onepunchman/images/8/85/Saitama_render.png/revision/latest/scale-to-width-down/248?cb=20191029171106&path-prefix=es";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el URL y el alt indicado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug()
    // expect(screen.getByRole("img").src).toBe(url)
    // expect(screen.getByRole("img").alt).toBe(title)
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
