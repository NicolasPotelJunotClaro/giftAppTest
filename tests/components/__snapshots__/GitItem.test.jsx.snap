// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GitItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One punch"
      src="https://static.wikia.nocookie.net/onepunchman/images/8/85/Saitama_render.png/revision/latest/scale-to-width-down/248?cb=20191029171106&path-prefix=es"
    />
    <p>
      One punch
    </p>
  </div>
</div>
`;
